src/Esfa.Vacancy.Register.Api/Orchestrators/SearchApprenticeshipVacanciesOrchestrator.cs

```diff
@@ -22,13 +23,19 @@ public SearchApprenticeshipVacanciesOrchestrator(IMediator mediator, IMapper map
         }
 
         public async Task<SearchResponse<ApprenticeshipSummary>> SearchApprenticeship(
-            SearchApprenticeshipParameters apprenticeSearchParameters)
+            SearchApprenticeshipParameters apprenticeSearchParameters, UrlHelper urlHelper)
         {
             if (apprenticeSearchParameters == null) ThrowValidationException();
 
             var request = _mapper.Map<SearchApprenticeshipVacanciesRequest>(apprenticeSearchParameters);
             var response = await _mediator.Send(request);
             var results = _mapper.Map<SearchResponse<ApprenticeshipSummary>>(response);
+
+            foreach (ApprenticeshipSummary summary in results.Results)
+            {
+                summary.ApiDetailUrl = urlHelper.Link("GetApprenticeshipVacancyByReference", new { vacancyReference = summary.VacancyReference });
```

Comment:
As mentioned above this should be done in the controller. Also we could create a constants file with RouteNames we use and reference those.

Reply:
This is the place for the mapping to happen
A RouteNames constant file has been added